PythonVisualizations/Trees.py

```diff
@@ -1,538 +1,405 @@
+
     def __promoteSuccessor(self, parent, node, direction):
-        successor = node.rightChild
+        successor = self.getRightChild(node)
+        self.highlightCircle(successor)
         newParent = node
                 
         #hunt for the right child's most left child
-        while successor.leftChild:
+        self.highlightLeftLine(successor)
+        self.wait(0.2)
+        self.unHighlightCircle(successor)
+        while self.getLeftChild(successor):
             newParent = successor
-            successor = successor.leftChild
-
-        #move the correct key to the deleted node's slot
-        node.key = successor.key
-
-        self.__reDraw(node, parent, direction)
-        self.__delete(newParent, successor, direction)
-
-    # ANIMATION METHODS
-    def speed(self, sleepTime):
-        return (sleepTime * (scaleDefault + 50)) / (scale.get() + 50)
-
-    ######For testing purposes
-    def printTree(self):
-        self.pTree(self.__root, "ROOT:  ", "")
-        print()
-
-    def pTree(self, n, kind, indent):
-        print("\n" + indent + kind, end="")
-        if n:
-            print(n, end="")
-            if n.leftChild:
-                self.pTree(n.leftChild, "LEFT:  ", indent + "    ")
-            if n.rightChild:
-                self.pTree(n.rightChild, "RIGHT:  ", indent + "    ")
-
-def clickInsert():
-    entered_text = textBox.get()
-    if entered_text:
-        val = int(entered_text)
-        if val < 100:
-            tree.insertElem(int(entered_text))
-        else:
-            outputText.set("Input value must be an integer from 0 to 99.")
-    textBox.delete(0, END)
-
-def clickFind():
-    entered_text = textBox.get()
-    if entered_text:
-        val = int(entered_text)
-        if val < 100:
-            found = tree.find(int(entered_text))
-            outputText.set("Found!" if found else "Value not found")
+            successor = self.getLeftChild(successor)
+            self.wait(0.2)
+            self.unHighlightLeftLine(newParent)
+            self.highlightLeftLine(successor)
+
+        self.wait(0.2)
+        self.unHighlightLeftLine(successor)
+        self.highlightCircle(successor)
+        self.wait(0.2)
+        self.unHighlightCircle(successor)
+
+        # delete the node that is to be deleted
+        self.removeNodeInternal(node)
+
+        # copy the items associated with successor
+        # 1. generate a new tag
+        newTag = self.generateTag()
+        # 2. copy the circle and the value
+        circle = self.copyCanvasItem(successor.drawable.display_shape)
+        circle_text = self.copyCanvasItem(successor.drawable.display_val)
+        # 3. add the appropiate tags
+        self.canvas.itemconfig(circle, tags=(newTag))
+        self.canvas.itemconfig(circle_text, tags=(newTag))
+        # 4. create a drawable
+        successorCopyDrawable = drawable(successor.getKey(), self.canvas.itemconfigure(circle, 'fill'), *(circle, circle_text))
```

Comment:
The call to `.itemconfigure(circle, 'fill')` needs a `[-1]` applied to it to get the last element of the tuple returned by that call.

Reply:
change implemented in commit 9eff8cf1db146e971c3a385aa7a81b5833fff71c